In [1]:
import numpy as np
import random
import pandas as pd

# Sonstige Bibliotheken

In [2]:
from tensorflow.keras.preprocessing.text import Tokenizer
from tensorflow.keras.preprocessing.sequence import pad_sequences

In [3]:
data = pd.read_csv('../Data/Shakespeare_data.csv')
training_data = data['PlayerLine'].str.lower()

# Tokenizer

In [4]:
#Tokenizer erstellen und fitten
tokenizer = Tokenizer()
tokenizer.fit_on_texts(data['PlayerLine'])

#Wörter in Integer umwandeln
sequences = tokenizer.texts_to_sequences(data['PlayerLine'])

#Maximale länge bestimmen (Wichtig für die Layer!)
max_sequence_length = max(len(seq) for seq in sequences)

#Mithilfe von 0 alle sequenzen auf dieselbe länge bringen
sequences = pad_sequences(sequences, maxlen=max_sequence_length, padding='post')

vocab_size = len(tokenizer.word_index) + 1

print(sequences[:3])

[[ 315    3    0    0    0    0    0    0    0    0    0    0    0    0
     0    0    0    0    0    0    0    0    0    0    0    0    0    0
     0    0    0    0    0    0    0    0    0    0    0    0    0    0
     0    0    0    0    0    0    0    0    0    0    0    0    0    0
     0    0    0    0    0    0    0    0    0    0    0    0    0    0
     0    0    0    0    0    0    0    0    0    0    0    0    0    0
     0    0    0    0    0    0    0    0    0    0    0    0    0    0
     0    0    0    0    0    0    0    0    0    0    0    0    0    0
     0    0    0    0    0    0    0    0    0    0    0    0    0    0
     0    0    0    0    0    0    0    0    0    0    0    0    0    0
     0    0    0    0    0    0    0    0    0    0    0    0    0    0
     0    0    0    0    0    0    0    0    0    0    0    0    0]
 [ 135    3  802    1  571    0    0    0    0    0    0    0    0    0
     0    0    0    0    0    0    0    0    0    0    0    0    0  

In [5]:
total_words_sum = len(tokenizer.word_index)+1
print(total_words_sum)

max_seq_len = max(len(x) for x in sequences)
print(max_seq_len)

25576
167


total_words_sum = len(tokenizer.word_index)+1
print(total_words_sum)

max_seq_len = max(len(x) for x in sequences)
print(max_seq_len)

25576
167

# Generator class

In [ ]:
class GAN:
    def __init__(self):
        next()

# Training